akka-actor/src/main/scala/akka/util/ByteString.scala

```diff
@@ -448,20 +448,23 @@ object ByteString {
 
     override def dropRight(n: Int): ByteString =
       if (n <= 0) this
-      else {
-        val last = bytestrings.last
-        if (n < last.length) new ByteStrings(bytestrings.init :+ last.dropRight1(n), length - n)
-        else {
-          val remaining = bytestrings.init
-          if (remaining.isEmpty) ByteString.empty
-          else {
-            val s = new ByteStrings(remaining, length - last.length)
-            val remainingToBeDropped = n - last.length
-            s.dropRight(remainingToBeDropped)
-          }
-        }
+      else if (n >= length) ByteString.empty
+      else dropRight0(n)
+
+    private[akka] def dropRight0(n: Int): ByteString = {
```

Comment:
Thanks for pointing it out!!! Yes, we should hide it :)
I'm planning to use `dropRight(...)` or `take(...)` to improve `slice(...)` so I might make `take0(...)`  `private` (instead of _completely_ hiding it inside `take(...)`)
Anyway will fix it for sure :+1: